### PR TITLE
baseline + one profile

### LINKED ISSUE #93: COSE algorithms in the base profile
We provide a SHOULD-level requirement for using “commonly adopted algorithms”

We shall leave it as-is for the “base” (“partial”) profile, and describe the exact expectations for a “full” profile.

This should stay in the base profile (i.e., common to all derived full profiles)

### LINKED ISSUE #95: key identification (per profile)
In §12 we say:

> _“If the IAK is a raw public key, the Instance and Implementation ID claims are used (together with the kid in the COSE header, if present) to assist in locating the key used to verify the signature covering the CWT token“_

Which is slightly problematic in terms of interop because of the potential ambiguity: both? If so, which one is authoritative? Only one, if so which?

It’s easier if we drop kid and just say we do [§F.1.14 of EAT](https://www.ietf.org/archive/id/draft-ietf-rats-eat-21.html#appendix-F.1.4) “Claim-Based Key Identification”

This is profile dependent (e.g., a “certified public key” profile would use a different method)

### LINKED ISSUE #96: Base profile vs full profiles
Use the current definition as the “base profile” and derive two “full profiles” modelled on what TF-M provides:

* Symmetric (COSE Mac0 w/ HMAC256/256 or HMAC384/384 or HMAC512/512)
* Asymmetric (COSE Sign1 w/ ES256 or ES384 or ES512)

With the provision that a sender MUST support at least one alg, the receiver MUST support all algs (in a full profile).

The base profile only defines a `profile_id` “namespace” for derived profiles, which is never seen on the wire (e.g., `http://arm.com/psa/2.0.0`).

Each full profile would be given its own profile_id in the parent namespace, which is what gets serialised (e.g., `http://arm.com/psa/2.0.0/s+hmac` and `http://arm.com/psa/2.0.0/a+es`).

Other jurisdictions may have to define their own profiles if these two are not OK. E.g., for an hypothetical Chinese market that must use SM2 instead of ES256 => `http://arm.com/psa/2.0.0/a+sm2`).

### LINKED ISSUE #97: `eat_profile` URI structure
It has been pointed out that `http://arm.com/psa/2.0.0` may not be ideal as a base URI for the `eat_profile` claim:

* delegating profile definition outside the Arm corp boundary may have "branding" implications
* `http` URIs can be automatically de-referenced, causing unneeded traffic to Arm's website

One possible alternative that was briefly discussed uses [TAG URIs](https://www.rfc-editor.org/rfc/rfc4151#section-2.1).

For example, `tag:arm.com,2023:psa` or (better, more neutral from a branding perspective) `tag:psacertified.org,2023:psa`.

They would extend naturally using the "fragment" part, e.g.: `tag:arm.com,2023:psa#s+hmac`.

### PR DESCRIPTION
**Pull request by thomas-fossati**
This PR replaces #101 

It is on top of #102 , so you only need to focus on [15668eb](https://github.com/thomas-fossati/draft-psa-token/pull/103/commits/15668eb6f682555985ae9ec3244f46e4c39b1aa1) and [6c3db51](https://github.com/thomas-fossati/draft-psa-token/pull/103/commits/6c3db51a7779b639c1f3ee6684e1748f5f9869df).

Fix #93
Fix #95 
Fix #96 
Fix #97

### CODE REVIEW

**Comment by adrianlshaw** [comment]
Why is tfm mentioned in the profile string? I think this profile ought to be implementation agnostic

**Comment by thomas-fossati** [comment]
> Why is tfm mentioned in the profile string? I think this profile ought to be implementation agnostic

I basically ran out of imagination.  I'm very open to suggestions :-)

**Comment by adrianlshaw** [comment]
Ah ok. Well, I believe that given that this a PSA document there should be clear separation between architecture and implementation. My suggestion is to drop `#tfm` from the string and move all TF-M references into the **Implementation Status** section. It's a non trivial suggestion, so I will sleep on it :-)

**Comment by SimonFrost-Arm** [comment]
I think the changes work. 

To @adrianlshaw point on use of TFM, can we keep the current split but add some text around line 666 (...suitable for TF-M) to the effect that these implementation choices are likely to be common or, if more specific, likely to be only more restrictive in choice of COSE protection. The relevant naming can then be specific to whoever can produce that solution.

**Review by SimonFrost-Arm — COMMENTED** [review]
LGTM once we've resolved the discussion on #tfm label

**Review by mathias-arm — APPROVED** [review]

File: draft-tschofenig-rats-psa-token.md [hunk 1]
```diff
@@ -516,11 +516,13 @@ The Profile Definition claim encodes the unique identifier that corresponds to
...
-apply to its type:
+The EAT `eat_profile` (claim key 265) is used.
 
-* The URI encoding MUST be used.
-* The value MUST be `http://arm.com/psa/2.0.0`.
+The URI encoding MUST be used.
+
+The value MUST be `tag:psacertified.org,2023:psa#tfm` for the profile defined in {{sec-tfm-profile}}.
+
+Future profiles derived from the baseline PSA profile SHALL mint their unique value as described in {{sec-profile-uri-structure}}.
```
Comment (hannestschofenig) [inline draft-tschofenig-rats-psa-token.md:525]:
```suggestion
Future profiles derived from the baseline PSA profile SHALL create their unique value, as described in {{sec-profile-uri-structure}}.
```

File: draft-tschofenig-rats-psa-token.md [hunk 2]
```diff
@@ -608,6 +632,63 @@ A PSA token is always directly signed by the PSA RoT.  Therefore, a PSA
...
+| Endorsements | See {{sec-psa-endorsements}} |
+| Freshness | nonce or epoch ID based |
+| Claims | Those defined in {{sec-psa-claims}}. As per general EAT rules, the receiver MUST NOT error out on claims it doesn't understand. |
+{: #tbl-baseline-profile title="Baseline Profile"}
+
+## Profile TFM
+{: #sec-tfm-profile}
+
+This profile is suitable for devices using TrustedFirmware-M {{TF-M}}.
+
```
Comment (SimonFrost-Arm) [inline draft-tschofenig-rats-psa-token.md:667]:
```suggestion
This profile is appropriate for the code base implemented in TF-M and should apply for most derivative implementations. If an implementation changes the requirements described below then, to ensure interoperability, this profile should be identified by a new profile value. This includes a restriction of the profile to a subset of the COSE Protection scheme requirements.
```

**Review by SimonFrost-Arm — APPROVED** [review]
Suggested qualification text for the use of the #tfm naming

**Pull request merged**